helpdesk_mgmt/models/helpdesk_ticket.py

```diff
@@ -18,10 +18,18 @@ def _read_group_stage_ids(self, stages, domain, order):
         stage_ids = self.env["helpdesk.ticket.stage"].search([])
         return stage_ids
 
+    def _get_user_domain(self):
+        res = "[]"
+        if not self.team_id:
+            return res
+        return "[('id', 'in', self.user_ids)]"
```

Comment:
This won't work, self is not passed to the domain methods, at least not from my testing in Odoo v15.

To get this to work you'd need to do something like this.
```suggestion
    def _get_user_domain(self):
        res = []
        active_model = self.env.context.get("active_model", "helpdesk.ticket")
        active_id = self.env.context.get("active_id")
        if active_model == "helpdesk.ticket" and active_id:
            ticket = self.env[active_model].browse(active_id)
            if ticket and ticket.team_id:
                res = [('id', 'in', self.user_ids.ids)]
        return res
```
The domain is only getting picked up when the user_ids field is being read in the onchange method, this won't flow through to the frontend form view, so from a user perspective you can choose a user from the dropdown, but it's immediately deleted when the onchange fires. If you check the form view definition the domain attribute that would have been passed from the field is being set to `domain="[('share', '=', False)]"`.

Also I don't believe `self.user_ids` be evaluated within a string.

I'd suggest changing the `user_ids` field to a computed field having a fixed domain.